src/interfaces/swig/shogun.i

```diff
@@ -136,9 +159,13 @@ namespace shogun
 		Tag<float64_t> tag_float64(name);
 
 		if ($self->has(tag_int32))
-			$self->put(tag_int32, (int32_t)value);
+		{
+			PUT_ENUM_INT_DISPATCHER(tag_int32, (int32_t) value);
```

Comment:
ah these are smart, didnt see them before

Reply:
I didn't want to use a macro but swig didn't like the templated functions I wrote so I just went with this...

Reply:
I could move this to the library actually and then it is possible to this check in C++ as well?

Reply:
actually could be a bit of a pain to do it with templating...

Reply:
i think it is fine, we can do that if needed